src/App.tsx

```diff
@@ -147,57 +151,62 @@ function QueryForm({
   apiQueryURL: string;
   onSubmitQuery: (url: string) => void;
 }) {
-  const [node, setNode] = useState<string[] | string | null>(null);
+  const [node, setNode] = useState<FieldInputOption | FieldInputOption[] | null>([{label: 'All', id: 'allNodes'}]);
   const [minAge, setMinAge] = useState<string | null>(null);
   const [maxAge, setMaxAge] = useState<string | null>(null);
-  const [sex, setSex] = useState<string[] | string | null>(null);
-  const [diagnosis, setDiagnosis] = useState<string[] | string | null>(null);
+  const [sex, setSex] = useState<FieldInputOption | FieldInputOption[] | null>(null);
+  const [diagnosis, setDiagnosis] = useState<FieldInputOption | FieldInputOption[] | null>(null);
   const [isControl, setIsControl] = useState<boolean>(false);
-  const [minNumSessions, setMinNumSessions] = useState<string[] | string | null>(null);
-  const [assessmentTool, setAssessmentTool] = useState<string[] | string | null>(null);
-  const [imagingModality, setImagingModality] = useState<string[] | string | null>(null);;
+  const [minNumSessions, setMinNumSessions] = useState<string | null>(null);
+  const [assessmentTool, setAssessmentTool] = useState<FieldInputOption | FieldInputOption[] | null>(null);
+  const [imagingModality, setImagingModality] = useState<FieldInputOption | FieldInputOption[] | null>(null);
+  const [searchParams, setSearchParams] = useSearchParams();
```

Comment:
Might make sense to eventually refactor the repeating stuff into it's own type: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-aliases

wdyt?

Reply:
Good call